### PR TITLE
Fixed sound issue

### PR DESCRIPTION
**Pull request by igoreydman**
Fixed sound issue where video audio wouldn't play if the device is on silent.

### CODE REVIEW

**Comment by twostraws** [comment]
Hmm… I'm not sure about this one. If someone mutes their device, they might have done so for a reason. If they don't hear any sound because their device is muted, they can just flick a switch to unmute and move on.

If anyone has strong opinions on this please let us know, but right now I'm OK with the current behavior.

**Comment by mayurtolani** [comment]
I think @igoreydman makes sense. 

If the user is asking the video to to be played, He/She expects audio in it. It actually becomes annoying when there is no audio, you have to put your phone out from silent mode watch the video and then later remember to put it back in silent mode. 

Also If you see any famous apps like facebook, youtube or many others allow their videos to play the audio even in silent mode.

**Comment by twostraws** [comment]
OK, then let's push ahead with enabling sound regardless of the mute switch – I'll add a comment to the code now.

File: Unwrap/Activities/Learn/LearnCoordinator.swift [hunk 1]
```diff
@@ -55,6 +55,13 @@ class LearnCoordinator: Coordinator, Awarding, Skippable, AlertHandling, AnswerH
         guard let videoURL = Bundle.main.url(forResource: activeStudyReview.title.bundleName, withExtension: "mp4") else { return }
         let player = AVPlayer(url: videoURL)
 
+        // Plays sound when device is physically muted or silent
+        do {
+            try AVAudioSession.sharedInstance().setCategory(AVAudioSessionCategoryPlayback)
+        } catch {
+            //Didn't work
```
Comment (twostraws) [inline Unwrap/Activities/Learn/LearnCoordinator.swift]: Given that we don't actually do anything with the catch block, could this be called just using `try?` and drop the do/catch blocks entirely?

**Comment by AHIOS** [comment]
Hmm... from the UX perspective I don't think this is the best solution.
Why don't add an overlay (only when silent) to the video remembering to the user that the device is muted?
Hence the user could consciously decide if re-enable the sound flicking the switch, or, for example tapping the mute overlay, enabling the sound even if the device is mute like other apps do, as said before.

**Comment by twostraws** [comment]
Because there is no official API for detecting when the mute mode is enabled.

**Comment by AHIOS** [comment]
> Because there is no official API for detecting when the mute mode is enabled.

But the workaround seems not so dirty, and is based on public API measuring the time elapsed playing a sound with predefined length.

You will accept a PR, if I implement that? 

IMHO is still a better behaviour than ignoring the switch

**Comment by twostraws** [comment]
The problem with weird workarounds is that they might cause problems in the future. I think perhaps the simplest thing to do is show an alert the first time a video is played, saying something like "If you don't hear any audio, please check your volume is turned up and your mute switch is off." Thoughts?

**Comment by AHIOS** [comment]
> The problem with weird workarounds is that they might cause problems in the future. I think perhaps the simplest thing to do is show an alert the first time a video is played, saying something like "If you don't hear any audio, please check your volume is turned up and your mute switch is off." Thoughts?

Many users usually dont reads annoying alerts, with only the OK button, they wanto to go straight to the content.

But one more tap, only one time, seems a good compromise to me

**Comment by twostraws** [comment]
To keep things simple, I just made it play the sound automatically – if folks want to play a movie without audio, they can just turn the sound down.

**Pull request closed without merging**